### PR TITLE
Fuzzy finder: allow line number at the end of a filename

### PR DESCRIPTION
**Pull request by olafurpg**
Previously, it was not possible to navigate straight to a specific line number with the "Files" tab of the fuzzy finder. This commit adds the ability to specify a line number with a `:NUMBER` suffix. This feature makes it possible to copy-paste error messages that use the `FILE:LINE:COLUMN` format and quickly open the location with the fuzzy finder.

## Test plan

- New storybook test
- New unit tests for query parsing

### CODE REVIEW

**Review by vovakulikov — COMMENTED** [review]
@olafurpg I am still trying to unwrap this, but I think I got your idea about having two transformers for the query and for URL (strip line and column parameters from the actual query but then use it in the URL). 

The problem I have is that initially, if you try to type `query:line` operator in the fuzzy search on the home page, fuzzy search doesn't strip the line operator for actual query for the API call 

Screenshot 2023-07-18 at 13 10 30

I don't have a full understanding, but it seems like we can't ignore `param` argument with transformers in the `FuzzyQuery.ts`

```ts
 protected fuzzySearch(): FuzzySearch {
        return new CaseInsensitiveFuzzySearch(this.searchValues(), undefined)
}
```

**Comment by olafurpg** [comment]
@vovakulikov good catch! I only tested it with the repo-local fuzzy finder. I will fix this for the global file finder

**Comment by olafurpg** [comment]
@vovakulikov fixed in the latest commit

File: client/web/src/fuzzyFinder/CaseInsensitiveFuzzySearch.ts [hunk 1]
```diff
@@ -25,12 +24,16 @@ function sortByTiebreakers<T>(values: FzfResultItem<T>[], tiebreakers: Tiebreake
...
+    constructor(public readonly values: SearchValue[], private readonly params?: FuzzySearchConstructorParameters) {
         super()
         this.totalFileCount = values.length
     }
 
     public search(parameters: FuzzySearchParameters): FuzzySearchResult {
+        const query = this?.params?.transformer?.modifyQuery
+            ? this.params.transformer.modifyQuery(parameters.query)
+            : parameters.query
+        console.log({ query })
```
[the remark this answers is not in the document]
Comment (olafurpg) [reply client/web/src/fuzzyFinder/CaseInsensitiveFuzzySearch.ts:36]: Fixed!

**Review by vovakulikov — APPROVED** [review]
Sorry for the slow response; thanks for the fix, and thanks for the new query operator.

**Pull request merged**